server/src/main/java/org/elasticsearch/search/fetch/subphase/DocValueFieldsContext.java

```diff
@@ -18,23 +18,129 @@
  */
 package org.elasticsearch.search.fetch.subphase;
 
+import org.elasticsearch.Version;
+import org.elasticsearch.common.Nullable;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.XContent;
+import org.elasticsearch.common.xcontent.XContentParseException;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentParser.Token;
+
+import java.io.IOException;
 import java.util.List;
+import java.util.Objects;
 
 /**
  * All the required context to pull a field from the doc values.
  */
 public class DocValueFieldsContext {
 
-    private final List<String> fields;
+    public static final String USE_DEFAULT_FORMAT = "use_field_mapping";
+
+    /**
+     * Wrapper around a field name and the format that should be used to
+     * display values of this field.
+     */
+    public static final class FieldAndFormat implements Writeable {
+
+        /**
+         * Parse a {@link FieldAndFormat} from some {@link XContent}.
+         */
+        public static FieldAndFormat fromXContent(XContentParser parser) throws IOException {
+            Token token = parser.currentToken();
```

Comment:
Could we maybe use the `ObjectParser` or `ConstructingObjectParser` here since that is now the preferred way to parse JSON?

Reply:
my understanding was that it is tricky when you need to be able to handle either a string or an object